lib/rest_openshift.rb

```diff
@@ -5,8 +5,8 @@ module Rest
     module OpenShift
       extend Helper
 
-      def self.populate(path, base_opts, opts)
-        populate_common("/oapi/<oapi_version>", path, base_opts, opts)
+      def self.populate(path, base_opts, opts, type)
+        populate_common("/apis/#{type}.openshift.io/<api_version>", path, base_opts, opts)
```

Comment:
I thought that having one `populate` method makes more sense now. i.e. move the populate method to `rest.rb` and work for all APIs. as `type` the whole `something.openshift.io` or `something.k8s.io` can be specified.

But in the kube version I see `#{path}.k8s.io` vs `#{type}.openshift.io` here. Why is this difference? Why for k8s we need a separate parameter? Would be much nicer if we can unify these so that any new APIs added in the future can be supported.

Reply:
In k8s path and type always match while in OpenShift - not?

Reply:
That's a mistake, I'll fix that.

Reply:
@akostadinov, so the choice is having user to put just the `type` vs `something.openshift.io` or `something.k8s.io`.  I was going for the shorter string the user needs to write.  Also, the separation of `openshift` and `k8s`.  It's kind of a grey area...I just went with the shorter things to write.

Reply:
@akostadinov, I've updated the PR...although I'm sticking with the separate `populate` method to save some writing.

Reply:
`populate` method is only an internal helper so not much user writing stuff. I was thinking for merging k8s and OpenShift calls in same rules and use the full identifyer such that if we need to call something that isn't `k8s.io` or `openshift.io` we can just add without adding additional rules files.

Let me know your thought about this. Maybe it's unlikely to have such new APIs as well perhaps not a bad idea to have rules separation.

Reply:
I'm in favor of keep it separate if it's okay with you.